pkg/cmd/controller/csr.go

```diff
@@ -24,23 +22,28 @@ func RunCSRApproverController(ctx context.Context, controllerCtx *EnhancedContro
 		return true, err
 	}
 
-	selector := labels.NewSelector()
-	labelsRequirement, err := labels.NewRequirement(monitoringLabelKey, selection.Equals, []string{monitoringLabelValue})
-	if err != nil {
-		return true, err
+	subjectSelectors := map[string]labels.Selector{
+		"prometheus":     labels.NewSelector(),
```

Comment:
CN is expected to be `prometheus-k8s`, isn't it?

Reply:
`CN=system:serviceaccount:openshift-monitoring:prometheus-k8s` for certs issued to Prometheus Pods/SA

`CN=system:serviceaccount:openshift-monitoring:metrics-server` for certs issued to Metrics-server Pods/SA

Reply:
(CMO will be setting those once https://github.com/openshift/cluster-monitoring-operator/pull/2329/files is merged)

Reply:
Yes, but your subject is wrong as used below in `csr.NewServiceAccountApprover(),`  (missing `-k8s`). You cannot do what you're doing here.

Also, I'd rather you don't set up the selectors first and then loop through the map. In case there were other bugs below, we want to make sure the selectors are always correct.

Reply:
You're right, I'll fix that (was trying to avoid duplicating code)